prereise/gather/flexibilitydata/human_in_the_loop/demo/similar day summer label.ipynb

```diff
@@ -0,0 +1,325 @@
+       "    <tr>\n",
+       "      <th>75%</th>\n",
+       "      <td>2.500000</td>\n",
+       "      <td>84.602500</td>\n",
+       "      <td>31.500000</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>max</th>\n",
+       "      <td>5.600000</td>\n",
+       "      <td>100.000000</td>\n",
+       "      <td>37.000000</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "        Wind Speed  Relative Humidity  Temperature\n",
+       "count  1540.000000        1540.000000  1540.000000\n",
+       "mean      2.019448          72.954338    29.543831\n",
+       "std       0.797484          14.241159     2.439095\n",
+       "min       0.300000          38.940000    24.000000\n",
+       "25%       1.450000          61.650000    27.500000\n",
+       "50%       2.000000          73.180000    29.500000\n",
+       "75%       2.500000          84.602500    31.500000\n",
+       "max       5.600000         100.000000    37.000000"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Here, we will use the temperature's 75% & 50% percentile as thresholds for label creation\n",
+    "\n",
+    "weather.loc[(weather.index >= datetime(2017, 6, 10)) & (weather.index <= datetime(2017, 8, 26)) & \\\n",
+    "            (weather.index.hour >= 15) & (weather.index.hour <= 19)].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12e3d66f-9347-4b41-8098-86263950aa43",
+   "metadata": {},
+   "source": [
+    "This section reads in the flexibility data and goes through it day by day along with price and weather data. It calculates the flexibility by comparing the peak-hour time period to the non-peak time period and see if the difference is greater than 0.4.\n",
+    "\n",
+    "For price, $40/MWh was set as the threshold and for weather, temperature's 75% & 50% percentile points are used as thresholds.\n",
+    "\n",
+    "After stepping through each day in each user's flexibility file (remember each column is an individual day with the header being the date and there are totally 24 time steps.), the labels are concatenated at the end for cGAN. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "2cc736ae-5d69-4329-afba-3627f9619ab0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for file in os.listdir(summerFlexPath):\n",
+    "    temp = pd.read_csv(os.path.join(summerFlexPath, file))\n",
+    "    dates = temp.columns\n",
+    "    \n",
+    "    flex_label = np.array([])\n",
+    "    temperature_label = np.array([])\n",
+    "    price_label = np.array([])\n",
+    "    \n",
+    "    for i in temp.columns:\n",
+    "        flex_avg = temp[i].iloc[15:20].copy()\n",
+    "        flex_avg = flex_avg.mean()\n",
+    "        \n",
+    "        other_avg = (temp[i].loc[0:14]).mean()\n",
```

Comment:
Are we forgetting to include certain time steps in the `other_avg`?

Reply:
This was done on purpose, since the user might choose to continue curtailing the load after the 3-7pm range, so I did not include those periods when calculating the label.

Reply:
for i in temp.columns:
        flex_avg = temp[i].iloc[15:20].copy()
        flex_avg = flex_avg.mean()
        
        other_avg = (temp[i].iloc[0:15]).mean()
        
        delta = (other_avg - flex_avg)/other_avg

Reply:
Done.